packages/grid/x-grid/src/tests/sorting.XGrid.test.tsx

```diff
@@ -216,56 +215,52 @@ describe('<XGrid /> - Sorting', () => {
 
       const t0 = performance.now();
       fireEvent.click(header);
-      await waitFor(() =>
-        expect(document.querySelector('.MuiDataGrid-sortIcon')).to.not.equal(null),
-      );
+      await waitFor(() => expect(document.querySelector('.MuiDataGrid-sortIcon')).to.not.be.null);
       const t1 = performance.now();
       const time = Math.round(t1 - t0);
       expect(time).to.be.lessThan(300);
     });
+  });
 
-    it('should render maximum twice', async function test() {
-      let renderCellCount = 0;
-      let renderHeaderCount = 0;
-      const TestCasePerf = () => {
-        const [cols, setCols] = React.useState<GridColDef[]>([]);
-        const data = useData(10, 10);
-
-        React.useEffect(() => {
-          if (data.columns.length) {
-            const newColumns = [...data.columns];
-            newColumns[1].renderHeader = (params) => {
-              renderHeaderCount += 1;
-
-              return <span>{params.field}</span>;
-            };
-            newColumns[1].renderCell = (params) => {
-              if (params.id === 0) {
-                renderCellCount += 1;
-              }
-              return <span>{params.value}</span>;
-            };
-            setCols(newColumns);
-          }
-        }, [data.columns]);
+  it('should prune rendering on cells', () => {
+    let renderCellCount = 0;
 
-        return (
-          <div style={{ width: 300, height: 300 }}>
-            <XGrid columns={cols} rows={data.rows} />
-          </div>
-        );
-      };
+    function CounterRender(props) {
+      React.useEffect(() => {
+        if (props.value === 'Nike') {
+          renderCellCount += 1;
+        }
+      });
+      return props.value;
+    }
 
-      render(<TestCasePerf />);
-      const header = getColumnHeaderCell(2);
-      renderHeaderCount = 0;
-      renderCellCount = 0;
-      fireEvent.click(header);
-      await waitFor(() =>
-        expect(document.querySelector('.MuiDataGrid-sortIcon')).to.not.equal(null),
+    const columns = [
+      {
+        field: 'brand',
+        renderCell: (params) => <CounterRender value={params.value} />,
+      },
+    ];
+
+    function Test(props) {
+      return (
+        <div style={{ width: 300, height: 300 }}>
+          <XGrid
+            {...baselineProps}
+            columns={columns}
+            checkboxSelection
+            {...props}
+          />
+        </div>
       );
-      expect(renderHeaderCount).to.equal(2);
-      expect(renderCellCount).to.equal(0);
-    });
+    }
+
+    const { setProps } = render(<Test />);
+    expect(renderCellCount).to.equal(1);
+    const cell = getCell(1, 0);
+    cell.focus();
+    fireEvent.click(cell);
+    expect(renderCellCount).to.equal(2);
+    setProps({ extra: true });
+    expect(renderCellCount).to.equal(2);
```

Comment:
On the baseline commit of #1513, we have:

```suggestion
    const { setProps } = render(<Test />);
    expect(renderCellCount).to.equal(2);
    const cell = getCell(1, 0);
    cell.focus();
    fireEvent.click(cell);
    expect(renderCellCount).to.equal(3);
    setProps({ extra: true });
    expect(renderCellCount).to.equal(8);
```